pkg/controller/garbagecollector/patch.go

```diff
@@ -39,21 +39,27 @@ func deleteOwnerRefStrategicMergePatch(dependentUID types.UID, ownerUIDs ...type
 	return []byte(patch)
 }
 
-// TODO: remove this function when we can use strategic merge patch
```

Comment:
Do we not plan on supporting smp on cr? or do we just plan on always allowing json patch in the gc, even if we start supporting smp on cr

Reply:
We dropped the plan for supporting smp on cr ref https://github.com/kubernetes/kubernetes/issues/58414 and https://github.com/kubernetes/kubernetes/pull/56606